src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -1547,3 +1566,22 @@ void AppHost::_CloseRequested(const winrt::Windows::Foundation::IInspectable& /*
     const auto pos = _GetWindowLaunchPosition();
     _logic.CloseWindow(pos);
 }
+
+void AppHost::_AppInitializedHandler(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                     const winrt::Windows::Foundation::IInspectable& /*arg*/)
+{
+    // GH#11561: We're totally done being initialized. Resize the window to
+    // match the initial settings, and then call ShowWindow to finally make us
+    // visible.
+
+    LaunchMode launchMode{};
+    _initialResizeAndRepositionWindow(_window->GetHandle(), _proposedRect, launchMode);
+
+    int nCmdShow = SW_SHOW;

```

Comment:
You can use `SW_SHOWDEFAULT` here to respect the wishes of the OS.
> Sets the show state based on the SW_ value specified in the [STARTUPINFO](https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/ns-processthreadsapi-startupinfoa) structure passed to the [CreateProcess](https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-createprocessa) function by the program that started the application.

Reply:
Huh. Interesting. Might be related to the weirdness in #9053

Reply:
Yeah I think so.

Reply:
HOLY SHIT that totally fixes it.